client/my-sites/checkout/composite-checkout/use-prepare-product-for-cart.js

```diff
@@ -210,28 +209,39 @@ function useAddProductFromSlug( {
 			debug( 'waiting on products/plans fetch' );
 			return;
 		}
-		if ( ! product ) {
-			debug( 'there is a request to add a product but no product was found', productAlias );
+		if ( ! validProducts || validProducts.length < 1 ) {
+			debug(
+				'there is a request to add one ore more products but no product was found',
+				productAliasFromUrl,
+				validProducts
```

Comment:
Probably we don't need to log validProducts here because even for debugging, what could it be besides undefined or an empty array? Unless you had a use case I hadn't thought of.

Reply:
No good case. It put that when I was implementing this to debug an issue I was having and then I forgot to delete it.